types/logger/logger.go

```diff
@@ -35,6 +36,23 @@ type Context context.Context
 
 type logfKey struct{}
 
+// J marshals v as JSON and writes it to logf formatted with the annotation to make logtail
+// treat it as a structured log.
+func J(logf Logf, recType string, v interface{}) {
```

Comment:
Suggest s/J/JSON/. It's not that long, and it's a lot clearer.

Reply:
What happens when `v` is not a struct/map? I'd like to see some tests for at least a slice and a basic type.

Also, corner case, but what if `v` contains a field called `logtail`? (It's not impossible to imagine this happening legitimately, if you were logging from within the logger.)

Reply:
Wait, Logf is a type. How about a JSON method on logf?

`logf.JSON(1, "Hostinfo", hi)`

... with the verbosity level thrown in there too?

Reply:
SGTM, but in an ideal world, the verbosity level would be some named constant rather than a bare int, to make it more obvious at call sites what it means.

Reply:
BTW (in case anyone is thinking about it), I like the fact that we're using our own specified type name rather than the fully qualified Go type name. Unfortunately, Go type names are not stable over time ever since type aliases were introduced (see https://github.com/golang/go/issues/36345).

Reply:
It's not necessarily just a type name, though.

Think of cases like: ("FooBeforeLogin", foo) ... ("FooAfterLogin", foo).

Let's try this as-is and we can always refactor later since it's an internal detail not going over the wire.